### PR TITLE
refactor: refactor transformer to be more typescript way

### PR DESCRIPTION
**Pull request by ahnpnl**
- Refactor `index.ts` to be more TypeScript way by using class syntax and extend `TsJestTransformer`

- Enable strict option for `tsconfig`

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -11,42 +14,37 @@ function transpileToCommonJS(code: string) {
...
-    filename: descriptor.filename,
-    source: descriptor.template.content,
-    compilerOptions: {
-      bindingMetadata: bindings
+class VueTsJestTransformer extends TsJestTransformer implements Transformer {
+  process(input: string, _filePath: Config.Path, _jestConfig: Config.ProjectConfig, _transformOptions?: TransformOptions): TransformedSource | string {
+    const { descriptor } = parse(input)
+    let bindings: BindingMetadata | undefined
+    let scriptSetupResult: SFCScriptBlock
+    let output: TransformedSource | string = ''
```
Comment (ahnpnl) [inline src/index.ts:22]: not sure what to set here as default return for `process`. Is it possible that `descriptor.script` or `descriptor.scriptSetup` cannot be null/undefined ?

Comment (lmiller1990) [reply src/index.ts:22]: Yep, both can be `undefined`.

You can define a vue file like this:
```
<template>
  <div />
</template
```

Without script.

File: src/index.ts [hunk 2]
```diff
@@ -11,42 +14,37 @@ function transpileToCommonJS(code: string) {
...
-
-  let output = scriptSetupCode
-
-  // this is fairly dirty
-  if (templateCode.includes('exports.render = render;')) {
-    output += templateCode + ';exports.default = {...exports.default, render};'
-  } else {
-    output += templateCode + ';exports.default = {...exports.default};'
+    // For js, ts, we can call super.process to ask ts-jest to compile. The rest we ask Vue compiler
+    // return super.process(input, filePath, jestConfig, transformOptions);
```
Comment (lmiller1990) [inline src/index.ts]: Is this commented out for a specific reason? Should we use this?

**Review by lmiller1990 — COMMENTED** [review]
Some questions/comments

File: src/index.ts (hunk 2 above)
Comment (ahnpnl) [reply src/index.ts]: ya that is the important one. If you want `ts-jest` to compile, then you need to deliver `ts` string into that `super.process`

**Review by lmiller1990 — APPROVED** [review]

**Pull request merged**